src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java

```diff
@@ -129,16 +162,39 @@ public SslSimpleBuilder setCipherSuites(String[] ciphersSuite) throws IllegalArg
         return this;
     }
 
-    public SslSimpleBuilder setClientAuthentication(SslClientVerifyMode verifyMode, String[] certificateAuthorities) {
-        if (isClientAuthenticationEnabled(verifyMode) && (certificateAuthorities == null || certificateAuthorities.length < 1)) {
-            throw new IllegalArgumentException("Certificate authorities are required to enable client authentication");
+    public SslSimpleBuilder setClientAuthentication(SslClientVerifyMode verifyMode) {
+        this.verifyMode = verifyMode;
+        return this;
+    }
+
+    public SslSimpleBuilder setCertificateAuthorities(String[] certificateAuthorities) {
+        if (certificateAuthorities == null || certificateAuthorities.length == 0){
+            throw new IllegalArgumentException("SSL certificate authorities is required");
         }
 
-        this.verifyMode = verifyMode;
         this.certificateAuthorities = certificateAuthorities;
         return this;
     }
 
+    public SslSimpleBuilder setTrustStore(String trustStoreType, String trustStoreFile, String trustStorePassword) throws Exception {
+        final Path trustStorePath = Paths.get(Objects.requireNonNull(trustStoreFile, "Trust Store path cannot be null"));
+        if (!Files.isReadable(trustStorePath)) {
+            throw new IllegalArgumentException(String.format("Trust Store file cannot be read. Please confirm the user running Logstash has permissions to read: %s", trustStoreFile));
+        }
+
+        this.trustStore = readKeyStore(
+                trustStorePath,
+                resolveKeyStoreType(trustStoreType, trustStorePath),
+                formatJksPassword(trustStorePassword)
+        );
+
+        if (!hasTrustStoreEntry(this.trustStore)) {
+            logger.warn("The provided Trust Store file does not contains any trusted certificate entry: {}. Please confirm this is the correct certificate and the password is correct", trustStoreFile);
```

Comment:
I wonder if here we should differentiate between our expectations from a key store being used as a trust store vs one doubling as both key store and trust store.
For a "trust store"-only p12 I'd expect that only certificates live in it, and any keys would be a mistake. on the other hand, when reusing a key store as a trust store anything goes, and should probably be discouraged in production.
I'd suggest checking how our other products do it, but in either case leaving here a comment saying that `hasTrustStoreEntry` could be made to verify either in a strict or loose modes.

Reply:
Overall I'm concerned that we let users get to a state that won't work. if we raised an exception here, now, I don't believe there would be any breaking environments that would be working before. We would have to, however, only raise exception if the client auth is set to required.

Reply:
>On the final part of the sentence, do we know of an invalid password situation that hits this code? from my test setting a wrong password in the trust store shows a different exception:

That message is indeed confusing. It's coming from the JDK when it tries to read the `truststore` file [here](https://github.com/logstash-plugins/logstash-input-http/pull/171/files/dd8e7d4e366ff0d9d30e4dc5792005bf5275f87b#diff-cd477b6c3d223bfe35f4076bec10e6240dbfaf88d5d714c91e90e80e93a93700R185) with an invalid password. I think I can catch this exception and re-throw it with a more meaningful message to the user. WDYT?

>Overall I'm concerned that we let users get to a state that won't work. if we raised an exception here, now, I don't believe there would be any breaking environments that would be working before. We would have to, however, only raise exception if the client auth is set to required.

Agreed, this concern also applies to other plugins, that currently have minimum to no validation regarding it. Considering it's a new functionally, I agree with you! It's better to avoid those invalid certificates from the very beginning!  I'll change it :)

Reply:
> That message is indeed confusing. It's coming from the JDK when it tries to read the truststore file [here](https://github.com/logstash-plugins/logstash-input-http/pull/171/files/dd8e7d4e366ff0d9d30e4dc5792005bf5275f87b#diff-cd477b6c3d223bfe35f4076bec10e6240dbfaf88d5d714c91e90e80e93a93700R185) with an invalid password. I think I can catch this exception and re-throw it with a more meaningful message to the user. WDYT?

just to clarify, I'm ok with the current exception that happens when the password is incorrect, but I don't think that the message we introduce in this PR is accurate by mentioning that the password may be incorrect at that point:

> logger.warn("The provided Trust Store file does not contains any trusted certificate entry: {}. Please confirm this is the correct certificate **and the password is correct**", trustStoreFile);

Reply:
Sorry, I misunderstood your message. I've changed it to throw an exception and removed the misleading part of the message. Thanks 👍